src/evotorch/neuroevolution/baseneproblem.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 NNAISENSE SA
```

Comment:
Why has this been added? Is there some future feature we think we will add here? Otherwise its conceptually cleaner just to have `NEProblem` etc. directly inherit from `Problem`.

Reply:
I agree that having `NEProblem` itself as the unifying base class for all the neuroevolution problem types is a clean concept. However, currently, `VecGymNE` does _not_ inherit from `NEProblem`: `VecGymNE` ended up doing things differently than `NEProblem` so building it upon `NEProblem` is not completely straight-forward. Since `VecGymNE` does not build upon `NEProblem`, `BaseNEProblem` serves as the new unifying base class for the problem types in the `neuroevolution` namespace.

Of course, it is an interesting and possible option to remove the gap between `VecGymNE` and `NEProblem` over time and then finally re-declare `VecGymNE` as a subclass of `NEProblem` (and then maybe `BaseNEProblem` can be dropped?). Worth considering.